src/Instrumentation/Laravel/src/Hooks/Hook.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Laravel\Hooks;
+
+use OpenTelemetry\API\Instrumentation\AutoInstrumentation\HookManagerInterface;
+use OpenTelemetry\API\Logs\LoggerInterface;
+use OpenTelemetry\API\Metrics\MeterInterface;
+use OpenTelemetry\API\Trace\TracerInterface;
+use OpenTelemetry\Contrib\Instrumentation\Laravel\LaravelConfiguration;
+
+interface Hook
+{
+    public function instrument(
+        HookManagerInterface $hookManager,
+        LaravelConfiguration $configuration,
+        LoggerInterface $logger,
+        MeterInterface $meter,
+        TracerInterface $tracer,
```

Comment:
Should I actually be passing the `*ProviderInterfaces` around?

My current understanding would be that I'd have to duplicate the instantiation or provide a Factory to give me back the implementations I want.

Reply:
Yes, I think you should. I also think you should just accept the `InstrumentationContext` here, and retrieve the providers you need for this particular hook (less rework if we add new providers in future). Also, I think the Tracer/Logger/etc for this hook should have a unique name that includes the component being instrumented, eg `io.opentelemetry.contrib.php.laravel.kernel` - the reason for this is that we can use [scope configuration](https://github.com/open-telemetry/opentelemetry-php/pull/1353) to turn off parts of auto-instrumentation based on that name.

Reply:
Thanks @brettmc, all of that sounds sensible. I will look to make the switch.

I will have to have a tinker with the scope configuration here and see how that goes!